### DISCUSSION TITLE
Retrieve all ASINs list

### DISCUSSION THREAD

**Discussion by therubioftw** [amzn/selling-partner-api-models#3160] [General]
I currently see I can request inventory stats, but that's only for FBA products.
Is there a way to ask for all current ASINs, regardless of status or fulfilment?

**Comment by yuri-lonesomelabs** [comment]
You need to generate a report of type `GET_MERCHANT_LISTINGS_ALL_DATA`

**Comment by therubioftw** [comment]
Cheers! I'm going to try that now.
I was actually trying to avoid using a report for this, since it's a slower process, but I guess it's the only option...
